src/main/java/practice/StreamPractice.java

```diff
@@ -48,20 +69,39 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * Example: select people of working age
      * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
      */
+
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
-    }
+        class FilterNecessaryPeople implements Predicate<Person> {
+            @Override
+            public boolean test(Person person) {
+                return (person.getAge() >= fromAge && person.getSex() == Person.Sex.MAN
+                        && person.getAge() <= maleToAge)
+                        || (person.getAge() >= fromAge && person.getSex() == Person.Sex.WOMAN
+                        && person.getAge() <= femaleToAge);
+            }
+        }
 
+        FilterNecessaryPeople filterNecessaryPeople = new FilterNecessaryPeople();
+        return peopleList.stream()
+                .filter(filterNecessaryPeople)
+                .collect(Collectors.toList());
+    }
     /**
      * Given a List of `Person` instances (having `name`, `age`, `sex` and `cats` fields,
      * and each `Cat` having a `name` and `age`),
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
+
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex() == Person.Sex.WOMAN
+                        && person.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
can we use only one `flatMap` operation here?
```suggestion
                .flatMap(...)
```

Reply:
> 
> 
> can we use only one `flatMap` operation here?

I dont think so.

Reply:
Viktor suggested and you so I changed after expenations